framework/di/Instance.php

```diff
@@ -126,7 +128,9 @@ public static function ensure($reference, $type = null, $container = null)
 
         if ($reference instanceof self) {
             $component = $reference->get($container);
-            if ($component instanceof $type || $type === null) {
```

Comment:
If `$type = null;` then instanceof return Error 
I wrote about this in https://github.com/yiisoft/yii2/issues/10467